File: frontend/src/types/eval.ts
export interface DatasetRow {
  id: string;
  dataset_id: string;
  question: string;
  context: string;
  expected_answer: string | null;
  created_at: string;
}

export interface Dataset {
  id: string;
  name: string;
  description: string | null;
  domain_tag: string | null;
  owner_id: string;
  created_at: string;
  row_count: number;
  rows?: DatasetRow[];
}

export interface ModelEndpoint {
  id: string;
  name: string;
  model_name: string;
  provider: string;
  base_url: string;
  context_window: number | null;
  cost_per_1k_input: number | null;
  cost_per_1k_output: number | null;
  is_judge: boolean;
  created_by: string;
  created_at: string;
}

export interface EvalConfig {
  sample_size?: number;
  concurrency: number;
  retry_count: number;
}

export interface EvalRunResponse {
  id: string;
  name: string;
  prompt_id: string;
  prompt_version_id: string;
  dataset_id: string;
  model_id: string;
  judge_model_id: string;
  status: string;
  config: EvalConfig;
  results_summary: Record<string, any> | null;
  created_by: string;
  created_at: string;
  completed_at: string | null;
}

export interface MetricScoreResponse {
  id: string;
  metric_name: string;
  score: number;
  explanation: string | null;
  judge_model_id: string;
  created_at: string;
}

export interface EvalResultResponse {
  id: string;
  dataset_row_id: string;
  output: string;
  latency_ms: number;
  input_tokens: number;
  output_tokens: number;
  estimated_cost: number;
  created_at: string;
  scores: MetricScoreResponse[];
}

export interface EvalRunDetailResponse extends EvalRunResponse {
  results: EvalResultResponse[];
}

File: backend/app/schemas/eval.py
from datetime import datetime
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict


# Dataset Schemas
class DatasetRowBase(BaseModel):
    question: str
    context: str
    expected_answer: Optional[str] = None


class DatasetRowResponse(DatasetRowBase):
    id: UUID
    dataset_id: UUID
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)


class DatasetBase(BaseModel):
    name: str
    description: Optional[str] = None
    domain_tag: Optional[str] = None


class DatasetCreate(DatasetBase):
    rows: List[DatasetRowBase] = []


class DatasetResponse(DatasetBase):
    id: UUID
    owner_id: UUID
    created_at: datetime
    row_count: int
    model_config = ConfigDict(from_attributes=True)


class DatasetDetailResponse(DatasetResponse):
    rows: List[DatasetRowResponse]
    model_config = ConfigDict(from_attributes=True)


# ModelEndpoint Schemas
class ModelEndpointBase(BaseModel):
    name: str
    model_name: str
    provider: str
    base_url: str
    context_window: Optional[int] = None
    cost_per_1k_input: Optional[float] = None
    cost_per_1k_output: Optional[float] = None
    is_judge: bool = False


class ModelEndpointCreate(ModelEndpointBase):
    api_key: str


class ModelEndpointResponse(ModelEndpointBase):
    id: UUID
    created_by: UUID
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)


# EvalRun Schemas
class EvalConfig(BaseModel):
    sample_size: Optional[int] = None
    concurrency: int = 5
    retry_count: int = 2


class EvalRunBase(BaseModel):
    name: str
    prompt_id: UUID
    prompt_version_id: UUID
    dataset_id: UUID
    model_id: UUID
    judge_model_id: UUID


class EvalRunCreate(EvalRunBase):
    config: EvalConfig = EvalConfig()


class EvalRunResponse(EvalRunBase):
    id: UUID
    status: str
    config: dict
    results_summary: Optional[dict] = None
    created_by: UUID
    created_at: datetime
    completed_at: Optional[datetime] = None
    model_config = ConfigDict(from_attributes=True)


class EvalRunListResponse(BaseModel):
    id: UUID
    name: str
    status: str
    created_at: datetime
    prompt_name: str
    model_name: str
    model_config = ConfigDict(from_attributes=True)


# EvalResult / Metric Schemas
class MetricScoreResponse(BaseModel):
    id: UUID
    metric_name: str
    score: float
    explanation: Optional[str] = None
    judge_model_id: UUID
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)


class EvalResultResponse(BaseModel):
    id: UUID
    dataset_row_id: UUID
    output: str
    latency_ms: float
    input_tokens: int
    output_tokens: int
    estimated_cost: float
    created_at: datetime
    scores: List[MetricScoreResponse]
    model_config = ConfigDict(from_attributes=True)


class EvalRunDetailResponse(EvalRunResponse):
    results: List[EvalResultResponse]
    model_config = ConfigDict(from_attributes=True)
